master/internal/db/postgres_experiments.go

```diff
@@ -526,25 +494,23 @@ WHERE t.id=$2
   AND s.metrics->'avg_metrics'->$1 IS NOT NULL
 ORDER BY batches;`, metricName, trialID, startBatches, endBatches, startTime)
 	if err != nil {
-		return nil, nil, maxEndTime, errors.Wrapf(err, "failed to get metrics to sample for experiment")
+		defer rows.Close()
+		return nil, errors.Wrapf(err, "failed to get metrics to sample for experiment")
 	}
-	defer rows.Close()
-	metricSeriesBatch, metricSeriesTime, maxEndTime = scanMetricsSeries(
-		rows, metricSeriesBatch, metricSeriesTime)
-	return metricSeriesBatch, metricSeriesTime, maxEndTime, nil
+	return rows, nil
 }
 
 // ValidationMetricsSeries returns a time-series of the specified validation metric in the specified
 // trial.
 func (db *PgDB) ValidationMetricsSeries(trialID int32, startTime time.Time, metricName string,
-	startBatches int, endBatches int) (metricSeriesBatch, metricSeriesTime []lttb.Point,
-	maxEndTime time.Time, err error,
+	startBatches int, endBatches int) (rows *sql.Rows, err error,
 ) {
-	rows, err := db.sql.Query(`
+	rows, err = db.sql.Query(`
 SELECT
   v.total_batches AS batches,
   (v.metrics->'validation_metrics'->>$1)::float8 AS value,
-  v.end_time as end_time
+  v.end_time as end_time,
+  v.metrics->>'avg_metrics' AS metrics
```

Comment:
I don't believe validation metrics have `avg_metrics`, this is probably supposed to be `v.metrics->>'validation_metrics'`.

also function signature and docs indicate that it's supposed to be selecting single metrics series, but in fact now it selects that series plus all metrics, which is strange.

Reply:
@ioga I suppose I would say the docs and implementation are still _technically_ true. We are selecting only one `metric series` to show to a use. I.e `validation_accuracy` or `training_accuracy` etc. However, since we had decided to store all future dependent variables, such as `epoch` as metrics themselves, this query now needs to return other independent variable/xaxis metrics as well. 

I think the docs and implementation are still true, it was just a bit _more_ true when we only returned a single metric vs batch. I am happy to add some docs or comments to describe why extra metrics are being returned.

Reply:
I suspect selecting both results in more data being transferred over the wire from postgres. if we're going to parse and unpack the metrics json in go code, it can be done for the "selected" metric just as well.

Reply:
@ioga just to clarify are you saying you'd rather in the queries itself i.e

SELECT
  total_batches AS batches,
  s.metrics->'avg_metrics'->$1 AS value,
  s.end_time as end_time
  s.end_time as end_time,
  s.metrics->>'avg_metrics' AS metrics
FROM trials t
  INNER JOIN steps s ON t.id=s.trial_id
WHERE t.id=$2

instead of returning all the metrics specifically this line

` s.metrics->>'avg_metrics' AS metrics`

we instead only query for the relevant x axis/independent variables that we know we care about?

Reply:
this was updated to instead remove `s.metrics->'avg_metrics'->$1 AS value` so that all metric parsing happens in the go code. This ended up being cleaner than updating current logic to handle a dynamic number of potential metric columns but if the current implementation does not suffice the alternative could be considered.